ui/src/components/pipeline/graph/Graph.tsx

```diff
@@ -119,8 +119,9 @@ export default function Graph(props: GraphProps) {
   const [edge, setEdge] = useState<Edge>();
 
   const handleEdgeClick = (event: MouseEvent, edge: Edge) => {
-    setEdgeOpen(true);
+    setEdge(edge);
```

Comment:
This is the correct way of solving the issue. We are passing the edge state to downstream